core/app/models/spree/payment.rb

```diff
@@ -113,13 +113,7 @@ def money
     alias display_amount money
 
     def amount=(amount)
-      self[:amount] =
-        case amount
-        when String
-          separator = I18n.t('number.currency.format.separator')
-          number    = amount.delete("^0-9-#{separator}\.").tr(separator, '.')
-          number.to_d if number.present?
-        end || amount
+      self[:amount] = Spree::LocalizedNumber.parse!(amount)
```

Comment:
There's this test:

    # rspec ./spec/controllers/spree/api/v1/payments_controller_spec.rb:121
    it "returns a 422 status when the amount is invalid" do
      payment.update_attributes(state: 'pending')
      api_put(:update, id: payment.to_param, payment: { amount: 'invalid' })
      expect(response.status).to be(422)
      expect(json_response['error']).to eql('Invalid resource. Please fix errors and try again.')
    end

So I guess we'll need a version of our `parse` method (What I called `parse!`) to be able to raise an exception or return `nil` instead of the previous default (0.0). Thinking about it again, having a method named `parse!` to raise an exception would make more sense. 

What do you think? Do you prefer having two methods, one safe `parse` method that always returns a BigDecimal and an unsafe `parse!` method that can raise an exception when the number is invalid, or you think it's better to change the controller logic so it accepts invalid numbers and default them to 0.0?